### PR TITLE
Bulk Load CDK: Unions to Disjoint Records

### PR DESCRIPTION
**Pull request by johnny-schmidt**
## What
Final mapper. I went a little light on the tests. The integration tests exercise this behavior pretty thoroughly.

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality for mapping union types to disjoint records, enhancing data loading processes.
- **Tests**
	- Added unit tests to validate the behavior of the new mapping functionality, ensuring robustness and correctness.

### CODE REVIEW

**Comment by johnny-schmidt** [comment]
@coderabbitai full review

File: airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/SchemalessTypesToJson.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.data
+
+import io.airbyte.cdk.load.data.json.toJson
+import io.airbyte.cdk.load.message.DestinationRecord
+import io.airbyte.cdk.load.util.serializeToString
+
+class SchemalessTypesToJson : AirbyteSchemaIdentityMapper {
```
Comment (edgao) [inline airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/SchemalessTypesToJson.kt:11]: this file + its test also looks like a dup from the previous PR? (SchemalessTypesToJsonString.kt)

**Review by edgao — APPROVED** [review]
had a question about the unknowntype thing, otherwise :shipit:

File: airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValueIdentityMapper.kt [hunk 2]
```diff
@@ -46,7 +46,10 @@ open class AirbyteValueIdentityMapper(
                 is TimestampTypeWithoutTimezone ->
                     mapTimestampWithoutTimezone(value as TimestampValue, path)
                 is NullType -> mapNull(path)
-                is UnknownType -> mapUnknown(value as UnknownValue, path)
+                is UnknownType -> {
+                    collectFailure(path)
+                    mapNull(path)
```
[the remark this answers is not in the document]
Comment (johnny-schmidt) [reply airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/AirbyteValueIdentityMapper.kt:51]: That was an accidental reversion during rebase. I'll fix it. As we've discussed, unknown types should be treated as wildcards and handled per-destination

File: airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/SchemalessTypesToJson.kt (hunk 1 above)
Comment (johnny-schmidt) [reply airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/data/SchemalessTypesToJson.kt:11]: rebase artifact

**Pull request merged**